src/extension.ts

```diff
@@ -183,43 +191,50 @@ export function activate(context: vscode.ExtensionContext) {
 			return;
 		}
 
+		const filesToRemove: string[] = [];
 		const clipboardContent = await vscode.env.clipboard.readText();
 		const clipboard = await writeFileOnDisk(clipboardContent);
+		filesToRemove.push(clipboard);
 
 		//by default compare clipboard against current file
 		let sameContent;
 		let current = editor.document.fileName;
 		const selection = editor.selection;
-		if (!selection.isEmpty) { 
+		if (!selection.isEmpty) {
 			//compare against current selection
 			const editorContent = editor.document.getText(selection);
 			current = await writeFileOnDisk(editorContent);
+			filesToRemove.push(current);
 			sameContent = await areFilesEqual([current, clipboard]);
 		} else if (editor.document.isUntitled) {
 			//compare against untitled file
 			const editorContent = editor.document.getText();
 			current = await writeFileOnDisk(editorContent);
+			filesToRemove.push(current);
 			sameContent = await areFilesEqual([current, clipboard]);
 		} else if (editor.document.isDirty) {
 			//compore against dirty content but invoke meld with current saved file
```

Comment:
I just saw it now as I developed on top of your changes.
Why do you compare the dirty content but invoke meld with the saved file? Is this also the behavior of the sublime extension?

For me this behavior is confusing.

If I run the command I would expect either to check **and** open the dirty version or check **and** open the saved version of the file.

Background of this question is that I want to add a settings value for the extension were you can select if generally the dirty version or the saved version should be used to compare. So I would make this also equal in the compare to clipboard function.

Reply:
Yes, as I remember this is also the behavior of the sublime extension. At first I was also confused.
But in my opinion this makes sense, because as a user I see the dirty content. If I compare with clipboard and the contents are the same I get informed there is no difference. So I'm normally just hitting the "save" button in VScode to confirm the unsaved changes. 
In the other case if the contents are different I open the original saved file with Meld, because ...
1. The original file is not temporary, thus not deleted after closing Meld. 
2. I can simply make my changes within Meld to the original file and hit the "save" button in Meld. Knowing exactly where the file gets stored. Returning to VScode after closing Meld, seeing the updated content. Using a temporary file would make the compare with clipboard feature really useless in this case.

If you really need to, please compare against the saved file right away. 


Your feature about comparing against dirty version and saved version. I woul limit that functionality to one file. 
For example you can compare saved File A with dirty File A. 
But not that it's possible to compare saved/dirty File B with dirty File A.

Reply:
thanks for the description.

Yes this behavior is ok for me.

I will think about it .. maybe a "diff to saved version" is really the best  option for this.